analyses/tmb-vaf-longitudinal/03-explore-TMB-mut.Rmd

```diff
@@ -0,0 +1,221 @@
+
+```{r create-stacked-barplot-filter, fig.width = 15, fig.height = 8, fig.fullwidth = TRUE}
+# Filter df and remove any samples with single timepoints
+df_plot_filter <- df %>%
+  filter(!tmb >= 10) %>%
+  unique() %>% 
+  arrange(Kids_First_Participant_ID, tumor_descriptor) %>%
+  group_by(Kids_First_Participant_ID) %>%
+  dplyr::summarise(tumor_descriptor_sum = str_c(tumor_descriptor, collapse = ";")) %>% 
+  filter(!tumor_descriptor_sum %in% c("Diagnosis", "Progressive", "Recurrence", "Second Malignancy", "Unavailable", "Deceased", "Progressive;Progressive")) %>% 
+  dplyr::left_join(df, by = c("Kids_First_Participant_ID", "tumor_descriptor_sum")) %>% 
+  mutate(tumor_descriptor = factor(tumor_descriptor),
+         tumor_descriptor = fct_relevel(tumor_descriptor, f)) %>% 
+  drop_na(tmb) 
+
+# length(unique(df_plot_filter$Kids_First_Participant_ID))
+
+# Define parameters for function
+ylim = max(df_plot_filter$tmb)
+df_plot_filter <- df_plot_filter
+
+# Run function
+fname <- paste0(plots_dir, "/", "TMB-genomic-no-hypermutants.pdf")
+print(fname)
+p <- create_stacked_barplot(tmb_df = df_plot_filter, ylim = ylim)
+pdf(file = fname, width = 25, height = 8)
+print(p)
+dev.off()
+```
+
+# TMB across timepoints and cancer types per Patient case
+We will explore TMB per cancer group over time by creating dumbbell plots. We classified by using cancer types with the highest number of samples (High- and Low-grade gliomas) versus any other cancer groups.
+
+```{r create-dumbbell-ct, fig.width = 12, fig.height = 8, fig.fullwidth = TRUE}
+# How many bs_samples per kids_id and cancer group?
+# print(table(df_plot_filter$cg_plot))
+print(df_plot_filter %>% 
+        count(cg_plot, Kids_First_Participant_ID))
+
+# Dumbbell plot per cancer group
+cancer_groups <- unique(as.character(df_plot_filter$cg_plot))
+cancer_groups <- sort(cancer_groups, decreasing = FALSE)
+print(cancer_groups)
+
+for (i in seq_along(cancer_groups)) {
+  print(i)
+  df_ct_sub <- df_plot_filter %>% 
+    filter(cg_plot == cancer_groups [i])
+  
+      if (i %in% c(3, 7, 8)) {
+    print(cancer_groups [i])
+    # Define parameters for function
+    ylim <- 2
+    } else if (i == 2) {
+      print(cancer_groups [i])
+      # Define parameters for function
+      ylim <- 6
+      } else {
+        print(cancer_groups [i])
+        # Define parameters for function
+        ylim <- 4
+      }
+    
+
+    # Name plots
+    fname <- paste0(plots_dir, "/", cancer_groups[i], "-TMB-dumbbell", ".pdf")
+    print(fname)
+    
+    # Run function
+    p <- create_dumbbell_ct(tmb_df = df_ct_sub, 
+                                 ylim = ylim, 
+                                 ct_id = cancer_groups[i])
```

Comment:
It looks like there are some patients with only one time point, but maybe the timepoints just have similar TMBs? In the High-grade glioma plot, for example.

Reply:
@rjcorb I am not seeing this in the HGG on my end.

![Screenshot 2023-10-03 at 6 49 40 PM](https://github.com/d3b-center/pbta-tumor-evolution/assets/46328680/c63957a0-1681-4cd2-9e2b-c17dde54eb16)

But I can see this for the `Low-grade glioma,Glial-neuronal tumor-TMB-dumbbell.pdf`.  Here, I confirm that both timepoints have the same TMB value.
![Screenshot 2023-10-03 at 6 50 15 PM](https://github.com/d3b-center/pbta-tumor-evolution/assets/46328680/7b9f7d6f-861f-4193-b69f-ff0cde30b90c)